engine-nextgen/src/main/kotlin/graphql/nadel/NextgenEngine.kt

```diff
@@ -134,6 +145,7 @@ class NextgenEngine @JvmOverloads constructor(
         service: Service,
         executionContext: NadelExecutionContext,
     ): ExecutionResult {
+        addTypenameToInterfacesAndUnions(topLevelField, typenameHintAlias, overallSchema)
```

Comment:
Is there a reason we're not doing this via a `NadelTransform`?

I see we're using `recursivelyRemoveKey` which just traverses the entire JSON tree, which we actually don't want because we end up traversing the entire tree.